service/src/main/java/tds/assessment/services/impl/AssessmentConfigSeedDataLoaderServiceImpl.java

```diff
@@ -0,0 +1,152 @@
+                                                     final TimeWindowRepository timeWindowRepository,
+                                                     final GradeRepository gradeRepository,
+                                                     final SubjectRepository subjectRepository,
+                                                     final LanguageRepository languageRepository,
+                                                     final AccommodationFamilyRepository accommodationFamilyRepository,
+                                                     final AssessmentGradeRepository assessmentGradeRepository) {
+        this.clientRepository = clientRepository;
+        this.configsDataCommandRepository = configsDataCommandRepository;
+        this.timeWindowRepository = timeWindowRepository;
+        this.gradeRepository = gradeRepository;
+        this.subjectRepository = subjectRepository;
+        this.languageRepository = languageRepository;
+        this.accommodationFamilyRepository = accommodationFamilyRepository;
+        this.assessmentGradeRepository = assessmentGradeRepository;
+    }
+
+    @Override
+    public void loadSeedData(final TestPackage testPackage) {
+
+        final String clientName = testPackage.getPublisher();
+        configsDataCommandRepository.copyTesteeAttributeSeedDataForClient(clientName);
+        configsDataCommandRepository.copyTesteeRelationshipAttributeSeedDataForClient(clientName);
+        clientRepository.save(new Client(clientName));
+
+        insertTimeWindowData(clientName);
+        insertClientGradeData(testPackage);
+
+        // Load the client subject and accommodation family
+        subjectRepository.save(new Subject(clientName, testPackage.getSubject()));
+        accommodationFamilyRepository.save(new AccommodationFamily(clientName, testPackage.getSubject()));
+    }
+
+    @Override
+    public void loadLanguage(final TestPackage testPackage) {
+        Set<Language> languages = testPackage.getAssessments().stream().flatMap(
+            assessment -> assessment.getSegments().stream().flatMap(segment -> {
+                    if (segment.getAlgorithmType().equals(Algorithm.FIXED_FORM.getType())) {
+                        return segment.segmentForms().stream()
+                            .flatMap(form -> form.itemGroups().stream()
+                                .flatMap(itemGroup -> itemGroup.items().stream()
+                                    .flatMap(item -> item.getPresentations().stream()
+                                        //TODO: do we need a presentation @label attribute?
+                                        .map(language -> new Language(testPackage.getPublisher(), language.getCode(), language.label()))
+                                    )
+                                )
+                            );
+                    } else if (segment.getAlgorithmType().contains("adaptive")) {
```

Comment:
Would it be possible to use `.equals(Algorithm.ADAPTIVE_2)` here instead of `.contains("adaptive")`?  Do we have adaptive selection algorithms that are not "adaptive2"?

Reply:
The legacy system accepts both "adaptive" and "adaptive2" as algorithm implementations. I think we may simply have to make it clear that "adaptive" and "fixedform" are currently the only acceptable values. We may have to programmatically map these to "adaptive2" in the TDS database during load though